src/main/java/org/mockbukkit/mockbukkit/ServerMock.java

```diff
@@ -1084,39 +1084,49 @@ public Recipe getRecipe(@NotNull NamespacedKey key)
 		return this.recipeManager.getRecipeByKey(RecipeType.CRAFTING, key);
 	}
 
-	@Nullable
 	@Override
-	public Recipe getCraftingRecipe(@NotNull ItemStack[] craftingMatrix, @NotNull World world)
+	public @Nullable Recipe getCraftingRecipe(@NotNull ItemStack[] craftingMatrix, @NotNull World world)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		Preconditions.checkArgument(craftingMatrix != null, "craftingMatrix must not be null");
+		Preconditions.checkArgument(craftingMatrix.length == 9, "craftingMatrix must be an array of length 9");
+		Preconditions.checkArgument(world != null, "world must not be null");
+		return this.recipeManager.getCraftingRecipe(craftingMatrix);
 	}
 
-	@NotNull
 	@Override
-	public ItemStack craftItem(@NotNull ItemStack[] craftingMatrix, @NotNull World world, @NotNull Player player)
+	public @NotNull ItemStack craftItem(@NotNull ItemStack[] craftingMatrix, @NotNull World world)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		@Nullable Recipe recipe = getCraftingRecipe(craftingMatrix, world);
+		if (recipe == null)
+		{
+			return ItemStack.empty();
+		}
+
+		return recipe.getResult();
 	}
 
 	@Override
-	public @NotNull ItemCraftResult craftItemResult(@NotNull ItemStack[] craftingMatrix, @NotNull World world)
+	public @NotNull ItemStack craftItem(@NotNull ItemStack[] craftingMatrix, @NotNull World world, @NotNull Player player)
```

Comment:
Player is unused here. There seems to be some more checks / behavior when a player is a parameter. I would probably just Make this method throw unimplemented operation exception, as there are alternative methods that does not have player related checks